### ISSUE TITLE
Weather results

### ISSUE THREAD

**Issue by julianfairfax** [searxng/searxng#1206]
**Is your feature request related to a problem? Please describe.**
Most search engines display a little card with the weather for a location when your request them to but SearXNG doesn't.

**Describe the solution you'd like**
Add a list of weather engines and show the results from them in a weather tab or an instant answer widget.

**Describe alternatives you've considered**
Show weather results from general search engines in an instant answer widget like on the search engine itself.

**Comment by return42** [comment]
SearXNG do not track user's location .. weather results from what locations?

**Comment by julianfairfax** [comment]
> SearXNG do not track user's location .. weather results from what locations?

From the location specified. Such as "weather in Paris"

**Comment by dalf** [comment]
Here a data structure that describe a weather result. It fit #1627 and #1728 ... but I may have forgot some fields and perhaps in the future we will have to change it a little to include new engines.

With this design, the weather results are display at the top of the results like DDG and Google do. Ping @liimee 

```python
from typing import List, Dict, Set, Union
from typing_extensions import TypedDict, NotRequired, Required, Literal
from datetime import datetime

class WeatherValues(TypedDict):
    temperature: float  # °C or °F (see values_imperial and values_metric below)
    feellike: float  # °C or °F
    wind_speed: int  # km/h or m/h
    visibility: int  # km or miles


class WeatherInfo(TypedDict):
    condition: str
    wind_direction: str
    humidity: int
    uvindex: int
    sunrise: int
    sunset: int
    values_imperial: WeatherValues
    values_metric: WeatherValues


class WeatherWhenAndInfo(TypedDict):
    date: datetime.date
    time: Union[str, datetime.time]
    tz: datetime.tzinfo
    info: WeatherInfo


class Weather(TypedDict):
    template: Literal["weather.html"]
    location: str
    currently: WeatherInfo
    forcasts: List[WeatherWhenAndInfo]
```

So the wttr.in engine could be look like (partial example):
```python
def response(resp):
    ....
    results = []
    results.append({
        'template': 'weather.html',
        'currently': {
                'condition': result["current_condition"][0]['weatherDesc'][0]['value'],
                'wind_direction': float(result["current_condition"][0]['winddirDegree']),
                ...
                'values_imperial': {
                    'feellike': result["current_condition"][0]['FeelsLikeF'],
                    'temperature': result["current_condition"][0]['temp_F'] 
                    ...
                },
                'values_metric': {
                    ...
                },
        },
        'forcasts' : [
            {
                ...
            },
            {
                ...
            },
            ...
        ],
    })
    return results
```

Then 
* `template.html` can safely display the various values, so the engines don't have to care about [XSS](https://en.wikipedia.org/wiki/Cross-site_scripting).
* additional feature: if the engine sets only the metric values, SearXNG computes the missing imperial values automatically (and vice versa).

**Comment by liimee** [comment]
looks good -- does that mean that `weather.html` template should be added?

**Comment by dalf** [comment]
> does that mean that weather.html template should be added?

Yes. So only `searxng/searx/templates/simple/result_templates/` has to implement the HTML rendering. We can even some CSS to help. Then the weather engines can follow the format described above.

**Comment by liimee** [comment]
I can't really create the template myself but if you'd like to create it I'm willing to have a look at it

**Comment by unixfox** [comment]
Weather data available more easily for Google engine with #1749, example taken from the results of the object type `WEATHER_ONEBOX`:
```json
"current": {
    "air_pressure_mbar": 1013.5,
    "air_pressure_tendency": "RISING",
    "chance_of_precipitation": 61,
    "condition": {
        "dark_theme_icon_index": 1,
        "icon_index": 0,
        "type": "CLEAR_WITH_PERIODIC_CLOUDS",
        "weather_icon": {
            "immersive_weather_icon_data": {
                "background_color": [
                    -7704080,
                    -7636501,
                    -9213727
                ]
            },
            "is_rainy": false,
            "is_snowy": false,
            "osrp_background_color": "5150A9"
        }
    },
    "dew_point_f": 58,
    "feels_like_temp_f": 72,
    "has_mandatory_fields": true,
    "heat_index_f": 72,
    "humidity": 60,
    "one_hour_liquid_precipitation_inches": 0,
    "one_hour_snowfall_inches": 0,
    "part_of_day": "NIGHTTIME",
    "precipitation_type": "RAIN",
    "temp_f": 72,
    "uv_index": 0,
    "visibility_miles": 8,
    "wind": {
        "direction": 310,
        "speed_mph": 7
    },
    "wind_chill_f": 72
},
```

Data for "weather paris"

**Comment by seniorm0ment** [comment]
Maybe a field could be added in options and the user could specify town and state and it can be saved as part of the search url listed under cookies. Of course this would only apply if they used that url.
Then the alternative would be weather paris.

I think having that setting is nice for convenience, and then if a user has that setting set to default weather texas and they search weather paris, it should override that setting and show weather for paris not texas. But if they type just weather, then it should show texas.
Or they can leave the setting blank for maximum privacy and just search weather paris. But at least you add the layer of convenience for users who want it
